### PR TITLE
add tools, fmt and lint targets to Makefile

### PR DESCRIPTION
**Pull request by supreeth7**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -35,22 +32,32 @@ build:
...
+.PHONY: tools
+tools:
+	@mkdir -p $(GOPATH)/bin
+	@ls $(GOPATH)/bin/ginkgo 1>/dev/null || (echo "Installing ginkgo..." && go get -u github.com/onsi/ginkgo/ginkgo@v1.16.4)
+	
 .PHONY: test
 test:
 	ginkgo -v -r tests
 
+getlint:
```
[the remark this answers is not in the document]
Comment (supreeth7) [reply Makefile:44]: I guess we can't do go get - https://golangci-lint.run/usage/install/#local-installation

[the remark this answers is not in the document]
Comment (supreeth7) [reply Makefile:44]: The go get installation is not recommended as stated [here](https://golangci-lint.run/usage/install/#install-from-source). So, I went ahead with the binary installation.

[the remark this answers is not in the document]
Comment (T0MASD) [reply Makefile:44]: thanks for the explanatios

[the remark this answers is not in the document]
Comment (T0MASD) [reply Makefile:44]:
```suggestion
# Installed using instructions from: https://golangci-lint.run/usage/install/#linux-and-windows
getlint:
```

**Pull request merged**